server/chat-plugins/helptickets.ts

```diff
@@ -352,6 +352,19 @@ export class HelpTicket extends Rooms.RoomGame {
 		// @ts-ignore
 		this.playerTable = null;
 	}
+	onChatMessage(message: string, user: User) {
+		// in case they namechange
+		if (this.room.auth.has(user.id) && user.connections.length) {
+			const roomids = message.match(/battle-(?:[a-z0-9]+)-(?:[0-9]+)(?:-[a-z0-9]+pw)?/ig);
```

Comment:
```suggestion
			const roomids = message.match(/battle-(?:[a-z0-9]+)-(?:[0-9]+)(?:-[a-z0-9]+pw)?/g);
```

Wait, why did you even bother listing `a-zA-Z` separately when you had an `i` flag? And no, room IDs are definitely case sensitive :[